src/libNode/MicroBlockPreProcessing.cpp

```diff
@@ -912,6 +1043,78 @@ void Node::UpdateBalanceForPreGeneratedAccounts() {
                         << counter);
 }
 
+void Node::StartTxnProcessingThread() {
+  LOG_MARKER();
+  auto t = [this]() -> void {
+    m_mediator.m_node->m_prePrepRunning = false;
+    if (m_mediator.ToProcessTransaction()) {
+      if (m_mediator.m_ds->m_mode == DirectoryService::Mode::IDLE) {
+        m_mediator.m_node->ProcessTransactionWhenShardBackup(
+            SHARD_MICROBLOCK_GAS_LIMIT);
+        if (!AccountStore::GetInstance().SerializeDelta()) {
+          LOG_GENERAL(WARNING, "AccountStore::SerializeDelta failed");
+        }
+      } else {
+        m_mediator.m_node->ProcessTransactionWhenShardBackup(
+            m_mediator.m_ds->m_microBlockGasLimit);
+      }
+    }
+  };
+  DetachedFunction(1, t);
+}
+
+bool Node::WaitUntilTxnProcessingDone() {
+  LOG_MARKER();
+  // wait for txn processing being ready by me (backup)
+  unique_lock<mutex> lock(m_mutexCVTxnProcFinished);
+  int timeout_time = std::max(
+      0,
+      ((int)MICROBLOCK_TIMEOUT -
```

Comment:
yes. formula should be still same. `TX_DISTRIBUTE_TIME_IN_MS` still exist. However, now we can lower it.
We better have this formula so that we can fine tune ( possibly reduce ) constants like `MICROBLOCK_TIMEOUT` and `TX_DISTRIBUTE_TIME_IN_MS` to target new block time.